### PR TITLE
33309: Eliminated AspectMock usage from TestGeneratorTest.php

### PR DESCRIPTION
**Pull request by anzin**
### Description
I've eliminated AspectMock usage from 
`dev/tests/unit/Magento/FunctionalTestFramework/Util/TestGeneratorTest.php`.

### Fixed Issues (if relevant)
1. Fixes magento/magento2#33309: [MFTF] Eliminate AspectMock from TestGeneratorTest (Complex!)

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/verification tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)
 - [ ] Changes to Framework doesn't have backward incompatible changes for tests or have related Pull Request with fixes to tests

### CODE REVIEW

**Review by andrewbess — DISMISSED** [review]
Hello @anzin 
Thank you for your contribution.
It looks well for me.

**Comment by jilu1** [comment]
@magento import pull request to https://github.com/magento-commerce/magento2-functional-testing-framework

**Comment by magento-engcom-team** [comment]
@jilu1 the pull request successfully imported.

**Comment by jilu1** [comment]
@anzin 
Thank You for your PR! Please address unit test failures in TestGeneratorTest.

**Review by bohdan-harniuk — APPROVED** [review]
Hello, @anzin!
Good job! Thank you for your contribution!

Hello, @jilu1!
Could you please check everything here? I've refactored the code a little bit. Now all tests passed successfully!

Thanks, Bohdan

**Comment by jilu1** [comment]
@magento import pull request to https://github.com/magento-commerce/magento2-functional-testing-framework

**Comment by magento-engcom-team** [comment]
@jilu1 the pull request successfully imported.

**Review by jilu1 — APPROVED** [review]

File: src/Magento/FunctionalTestingFramework/Util/Filesystem/CestFileCreatorUtil.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+     * If the _generated directory doesn't exist it will be created.
+     *
+     * @param string $filename
+     * @param string $exportDirectory
+     * @param string $testPhp
+     *
+     * @return void
+     * @throws TestFrameworkException
+     */
+    public function create(string $filename, string $exportDirectory, string $testPhp): void
```
Comment (KevinBKozan) [inline src/Magento/FunctionalTestingFramework/Util/Filesystem/CestFileCreatorUtil.php:53]: I like the idea of pulling some code out of TestGenerator, but I don't think that creating a singleton that hold no data for a single utility function makes sense.

I would sooner that they change the visibility of `createCestFile` to public and mock that method instead. This is a lot of code to introduce to mock a single method.

**Review by KevinBKozan — CHANGES_REQUESTED** [review]

File: src/Magento/FunctionalTestingFramework/Util/Filesystem/CestFileCreatorUtil.php (hunk 1 above)
Comment (bohdan-harniuk) [reply src/Magento/FunctionalTestingFramework/Util/Filesystem/CestFileCreatorUtil.php:53]: @KevinBKozan, as for me, in general, the composition is much better than big classes with a variety of functions.
@jilu1, what do you think about that case?

Comment (bohdan-harniuk) [reply src/Magento/FunctionalTestingFramework/Util/Filesystem/CestFileCreatorUtil.php:53]: cc. @andrewbess

Comment (KevinBKozan) [reply src/Magento/FunctionalTestingFramework/Util/Filesystem/CestFileCreatorUtil.php:53]: @bohdan-harniuk Yeah, I see your point in that. I'm just vigilant about creating any classes that will perpetually live in memory, but I suppose that's not much of a concern if there's not a large amount of data being held.

I'm happy to defer to @jilu1 's original review in this case, I'll approve this and we can merge it.

**Pull request merged**